### PR TITLE
feat: Add option Always Run

### PR DESCRIPTION
**Pull request by neko-para**
Add an option to allow run program on empty testcases.
Update zh_CN translations.

related issue: #408

## Type of changes

- [ ] Bug fix (changes which fix an issue)
- [ ] New feature (changes which add functionality)
- [ ] Documentation (changes which modify the documentation only)
- [ ] Style change (changes which do not affect the meaning of the code: code formatting, etc.)
- [ ] Refactor (changes which affect the meaning of the code but neither fix a bug nor add a feature)
- [ ] Performance improve (changes which improve performance)
- [ ] Test (changes which add tests)
- [ ] Build (Changes that affect the build system or external dependencies)
- [ ] CI (changes to CI configuration files and scripts)
- [ ] Chore (changes which do not belong to any type above)
- [ ] Revert (revert previous changes)

## Checklist

- [ ] I have read the [CONTRIBUTING](https://github.com/cpeditor/cpeditor/blob/master/CONTRIBUTING.md) document.
- [ ] I have tested these changes locally, and this fixes the bug/the new feature behaves as the expectation.
- [ ] The settings file in the old version can be used in the new version after this change.
- [ ] These changes only fix a single bug/introduces a single feature. (Otherwise, open multiple Pull Requests instead, unless these bugs/features are closely related.)
- [ ] The commit messages are clear and detailed. (Otherwise, use `git reset` and commit again, or use `git rebase -i` and `git commit --amend` to modify the commit messages.)
- [ ] These changes don't remove an existing feature. (Otherwise, add an option to disable this feature instead, unless it's necessary to remove this feature.)
- [ ] If there are changes of the text displayed in the UI, I have wrapped them in `tr()` or `QCoreApplication::translate()`.
- [ ] I have documented these changes in [CHANGELOG.md](https://github.com/cpeditor/cpeditor/blob/master/doc/CHANGELOG.md), or these changes are not notable.

### CODE REVIEW

**Comment by neko-para** [comment]
Currently I put the option under Action/General. Maybe there is a better place.

File: src/Settings/PreferencesWindow.cpp [hunk 1]
```diff
@@ -208,7 +208,7 @@ PreferencesWindow::PreferencesWindow(QWidget *parent) : QMainWindow(parent)
         .end()
         .page("Appearance", tr("Appearance"), new AppearancePage())
         .dir("Actions", tr("Actions"))
-            .page("General", tr("General"), {"Hot Exit/Enable"})
+            .page("General", tr("General"), {"Hot Exit/Enable", "Always Run"})
```
Comment (coder3101) [inline src/Settings/PreferencesWindow.cpp:211]:
```suggestion
            .page("General", tr("General"), {"Hot Exit/Enable", "Run on empty testcases"})
```

File: src/Settings/settings.json [hunk 2]
```diff
@@ -757,5 +757,11 @@
         "default": "default",
         "param": "Core::StyleManager::styleList()",
         "tip": "The style of the whole application."
+    },
+    {
+        "name": "Always Run",
```
Comment (coder3101) [inline src/Settings/settings.json:762]: Run for empty test cases

File: src/Settings/settings.json [hunk 3]
```diff
@@ -757,5 +757,11 @@
...
+        "desc": "Always run the program",
```
Comment (coder3101) [inline src/Settings/settings.json:763]:
```suggestion
        "desc": "Allow running your code on empty testcase",
```

File: src/Settings/settings.json (hunk 2 above)
Comment (coder3101) [inline src/Settings/settings.json:762]:
```suggestion
        "name": "Run on empty test cases",
```

File: src/mainwindow.cpp [hunk 4]
```diff
@@ -168,7 +168,12 @@ void MainWindow::run()
 
     for (int i = 0; i < testcases->count(); ++i)
     {
-        if (!testcases->input(i).trimmed().isEmpty() && testcases->isShow(i))
+        if (testcases->input(i).trimmed().isEmpty())
+        {
+            if (SettingsHelper::isAlwaysRun())
```
Comment (coder3101) [inline src/mainwindow.cpp:173]:
```suggestion
            if (SettingsHelper::isRunOnEmptyTestCases())
```

Member might be different that this.

File: src/mainwindow.cpp [hunk 5]
```diff
@@ -168,7 +168,12 @@ void MainWindow::run()
 
     for (int i = 0; i < testcases->count(); ++i)
     {
-        if (!testcases->input(i).trimmed().isEmpty() && testcases->isShow(i))
+        if (testcases->input(i).trimmed().isEmpty())
+        {
+            if (SettingsHelper::isRunOnEmptyTestcase())
+                run(i);
+        }
+        else if (testcases->isShow(i))
```
Comment (ouuan) [inline src/mainwindow.cpp:176]: Are you sure it should be run even if it's hidden?

**Pull request merged**